File: Logger/src/main/java/com/cyrillrx/android/logger/Severity.java
package com.cyrillrx.android.logger;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author Cyril Leroux
 *         Created on 20/10/15
 */
public class Severity {

    public static final int FATAL = 0;
    public static final int ERROR = 1;
    public static final int WARN = 2;
    public static final int INFO = 3;
    public static final int DEBUG = 4;
    public static final int VERBOSE = 5;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({FATAL, ERROR, WARN, INFO, DEBUG, VERBOSE})
    public @interface LogSeverity { }
}

File: logger/logger/src/commonMain/kotlin/com/cyrillrx/logger/Severity.kt
package com.cyrillrx.logger

import kotlin.experimental.ExperimentalObjCName
import kotlin.native.ObjCName

/**
 * @author Cyril Leroux
 *         Created on 20/10/2015.
 */
enum class Severity(val level: Int, val label: String, val emoji: String) {
    FATAL(0, "FATAL", "💥"),
    ERROR(1, "ERROR", "❌"),
    WARN(2, "WARN", "⚠️"),
    INFO(3, "INFO", "ℹ️"),

    @OptIn(ExperimentalObjCName::class)
    @ObjCName("_DEBUG")
    DEBUG(4, "DEBUG", "🐞"),
    VERBOSE(5, "VERBOSE", "🔍"),
    ;
}
